engines/advancedDetector.cpp

```diff
@@ -392,6 +400,16 @@ Common::Error AdvancedMetaEngineDetection::createInstance(OSystem *syst, Engine
 		return Common::kNoGameDataFoundError;
 
 	DetectedGame gameDescriptor = toDetectedGame(agdDesc);
+	ADGameDescription *desc = (ADGameDescription *)new byte[sizeof(ADGameDescription)+128];
```

Comment:
since we know the length of the strings inside the struct, we could allocate it all in a single allocation, but if there's any pointers in the subclass then that would be a problem